src/main/java/com/ibm/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -252,8 +268,9 @@ public IResultsProvider call() throws AbortException {
 				try {
 					setInstallDir();
 		    		scan.run();
-		    		IResultsProvider provider = scan.getResultsProvider();
 		    		
+                                IResultsProvider provider=new NonCompliantIssuesResultProvider(scan.getScanId(), scan.getType(), scan.getServiceProvider(), scan.getProgress());
```

Comment:
Instead of scan.getProgress() can you use the IProgress we already have in this function?  That would negate the need to add the getProgress() method to the IScan interface.